Kernel/Process.cpp

```diff
@@ -556,6 +557,7 @@ siginfo_t Process::reap(Process& process)
 #endif
     ASSERT(process.is_dead());
     g_processes->remove(&process);
+    process.m_next = process.m_prev = nullptr; // mark it as reaped
```

Comment:
Wait, `InlineLinkedList` doesn't clear out the next/prev pointers when removing something?
Let's fix that in `InlineLinkedList` instead of patching the pointers here :)